### ISSUE TITLE
Internal services incorrectly require a port to be defined

### ISSUE THREAD

**Issue by FuNK3Y** [traefik/traefik#7821]
<!-- PLEASE FOLLOW THE ISSUE TEMPLATE TO HELP TRIAGE AND SUPPORT! -->

Bug

<!--

Internal services require a port to be defined, which does not make sense

-->

### What did you do?
I migrated the following traefik from a bridge network to a macvlan one

<!--

HOW TO WRITE A GOOD BUG REPORT?

- Respect the issue template as much as possible.
- The title should be short and descriptive.
- Explain the conditions which led you to report this issue: the context.
- The context should lead to something, an idea or a problem that you’re facing.
- Remain clear and concise.
- Format your messages to help the reader focus on what matters and understand the structure of your message, use Markdown syntax https://help.github.com/articles/github-flavored-markdown

-->

### What did you expect to see?
The dashboard should have been available under traefik.mydomain.com


### What did you see instead?
Error message `port is missing`


### Output of `traefik version`: (_What version of Traefik are you using?_)

<!--
`latest` is not considered as a valid version.

For the Traefik Docker image:
    docker run [IMAGE] version
    ex: docker run traefik version

-->

```
Version:      2.4.0                                                                                                                                                                                                                                                 
Codename:     livarot                                                                                                                                                                                                                                               
Go version:   go1.15.6                                                                                                                                                                                                                                              
Built:        2021-01-19T17:26:51Z                                                                                                                                                                                                                                  
OS/Arch:      linux/amd64   
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```yaml
version: "2.0"
networks:
  macvlan:
    driver: macvlan
    driver_opts:
      parent: interface
    ipam:
      config:
      - subnet: 192.168.2.0/24
        ip_range: 192.168.2.0/24
        gateway: 192.168.2.254
services:
  traefik:
    image: traefik
    networks:
      - macvlan
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    restart: always
    command:
      - --api
      - --providers.docker=true
    labels:
      - traefik.enable=true
      - traefik.http.routers.traefik.middlewares=auth
      - traefik.http.routers.traefik.entrypoints=websecure
      - traefik.http.routers.traefik.rule=Host(`traefik.mydomain.com`)
      - traefik.http.routers.traefik.service=api@internal
```

<!--
Add more configuration information here.
-->
The problem is certainly on that line: https://github.com/traefik/traefik/blob/326be29568426e4ac8e2bf65fdae3906e8dd58fd/pkg/provider/docker/config.go#L269 
This condition should not apply to internal services.

Adding `- traefik.http.services.traefik.loadbalancer.server.port=443` to my config is a working workaround

**Comment by jbdoumenjou** [comment]
Hello @FuNK3Y,

Could you provide our Traefik log from the beginning and the associated configuration?

**Comment by FuNK3Y** [comment]
@jbdoumenjou 
Here is the complete log:
 ```
 traefik_1  | time="2021-01-30T10:14:59Z" level=info msg="Configuration loaded from flags."
traefik_1  | time="2021-01-30T10:14:59Z" level=error msg="service \"traefik-traefik-debug\" error: port is missing" providerName=docker container=traefik-traefik-debug-7cbebd50f934c65ed54ddb5c2cae7d2bd1ad461dc68293442894cf5574a09d3c
```

**Comment by scottwallacesh** [comment]
I'm also seeing this but with a standard bridge network.
```yaml
networks:
  traefik:
    internal: true
    name: traefik_internal
    ipam:
      config:
        - subnet: 10.0.3.0/24
```

**Comment by spali** [comment]
It works when not defining a rule and just rely on the default rule or when making the network not `internal`:
```yaml
services:
  traefik:
    labels:
      - "traefik.enable=true"
      #- "traefik.http.routers.traefik.rule=Host(`traefik.${DOMAIN_NAME}`)"
      - "traefik.http.routers.traefik.service=api@internal"

networks:
  proxy:
    name: proxy
    internal: true
```
works with the default rule.

But as soon as I overwrite the default rule with a label:
```yaml
services:
  traefik:
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik.rule=Host(`traefik.${DOMAIN_NAME}`)"
      - "traefik.http.routers.traefik.service=api@internal"

networks:
  proxy:
    name: proxy
    internal: true
```
I get:
```
time="2021-11-05T12:34:04+01:00" level=error msg="service \"traefik-proxy\" error: port is missing" container=traefik-proxy-e29d2960a744b1e66173b05865fe51a72569be1e754e5c353240c8c93705b15f providerName=docker
```

And:
```yaml
services:
  traefik:
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik.rule=Host(`traefik.${DOMAIN_NAME}`)"
      - "traefik.http.routers.traefik.service=api@internal"

networks:
  proxy:
    name: proxy
    #internal: true
```
works.

So to me it looks like to be a combination of an `internal` only network with a rule on the container.

**Comment by austinsasko** [comment]
Also experienced this bug in the same manner as the above

**Comment by vysmaty** [comment]
This bug still exists in v2.11.2.
_Thank you @spali for provided information._

**Comment by nmengin** [comment]
I closed this issue in favor of #5732 which deals with the same issue.
